mesonbuild/dependencies.py

```diff
@@ -644,13 +644,19 @@ def detect_win_root(self):
 
     def get_compile_args(self):
         args = []
+        include_dir = ''
         if self.boost_root is not None:
             if mesonlib.is_windows():
-                args.append('-I' + self.boost_root)
+                include_dir = self.boost_root
             else:
-                args.append('-I' + os.path.join(self.boost_root, 'include'))
+                include_dir = os.path.join(self.boost_root, 'include')
         else:
-            args.append('-I' + self.incdir)
+            include_dir = self.incdir
+        # using "-isystem" on these default include dirs causes "havoc"
```

Comment:
This comment is too vague; no one is going to remember a month from now what this "havoc" was and why this hack is needed. Please explain the reasons for this hack, perhaps with links for more information.